### DISCUSSION TITLE
Dice Metric args are unclear

### DISCUSSION THREAD

**Discussion by sh3rlock14** [Lightning-AI/torchmetrics#1458] [Classification] [answered]
Hi there!
I was implementing a 3D volume segmentation model and hence I wanted to use torchmetrics Dice Score...
Alas, when I tried to use Dice I discovered that by setting some args, such as `average = None`, even if the docs let intend that is allowed, it is not!
Indeed: at row 140 there's:
`average: Optional[Literal["micro", "macro", "weighted", "none"]] = "micro",`,

but at row 166 we have: 
```
if average not in ["micro", "macro", "samples"]:
    raise ValueError(f"The `reduce` {average} is not valid.")
```

I was also wondering if your Dice support the `preds` and the `g_truth` to be in "one hot format" (not one hot encoding: I am in a multi-class, multi-label scenario, so I can have more than one "1" along C dim).

Thanks in advance for your help 🙏🏻 !
Mattia

**Comment by blazejdolicki** [comment]
I also stumbled upon the same issue, I think this is a bug in the library, see my comment: https://github.com/Lightning-AI/metrics/issues/1425#issuecomment-1400329109

**Comment by Borda** [comment, marked as answer]
@sh3rlock14 @blazejdolicki — this has been resolved by a complete API overhaul. The old `Dice` metric with confusing `average`/`mdmc_average` params has been replaced.

**Current API (v1.9.0) — use `torchmetrics.segmentation.DiceScore`:**
```python
from torchmetrics.segmentation import DiceScore

# For 3D volume segmentation (N, C, D, H, W) one-hot format:
dice = DiceScore(
    num_classes=5,
    average="macro",            # "micro", "macro", "weighted", "none"
    input_format="one-hot",     # expects (N, C, ...) one-hot encoded
    include_background=False,   # skip background class
    per_class=False,            # True to get per-class scores
)

score = dice(preds, target)
```

**For multi-label / overlapping classes** (multiple "1"s along channel dim):
`input_format="one-hot"` handles this correctly — each channel is an independent binary mask:
```python
dice = DiceScore(
    num_classes=C,
    average="none",
    input_format="one-hot",
    include_background=False,
)
per_class_dice = dice(preds.long(), target.long())
```

The old `average="none"` vs `average="samples"` confusion is gone — standard naming now, consistent with scikit-learn.

Docs: [DiceScore](https://lightning.ai/docs/torchmetrics/stable/segmentation/dice.html)
